programs/openbook-v2/src/instructions/consume_events.rs

```diff
@@ -66,17 +70,24 @@ pub fn consume_events(ctx: Context<ConsumeEvents>, limit: usize) -> Result<()> {
     let mut event_queue = ctx.accounts.event_queue.load_mut()?;
     let remaining_accs = &ctx.remaining_accounts;
 
-    // Iterate over event_queue
-    for _ in 0..limit {
-        let event = match event_queue.peek_front() {
-            None => break,
+    let slots_to_consume: Vec<usize> = match slots {
+        Some(slots) => slots.into_iter().take(limit).collect(),
+        None => event_queue
+            .iter()
+            .take(limit)
```

Comment:
Iterating over the slots isn't what costs lots of CU. I think it'd be fine to go with all slots here and use up `limit` only when an actual event processing happens.

Reply:
(otoh you should get in trouble with borrows then, so probably not worth the effort)

Reply:
yeah, i tried to use a closure like `let helper = |it: Box<dyn Iterator<Item = usize>>| it;`, but the borrow checker doesn't like it